drivers/azure/azureutil/azureutil.go

```diff
@@ -520,6 +532,33 @@ func (a AzureClient) CreateVirtualMachine(resourceGroup, name, location, size, a
 	return err
 }
 
+func (a AzureClient) CreateVirtualMachineExtension(os, resourceGroup, name, location string) error {
+	vmExtension := compute.VirtualMachineExtension{
+		Type:     to.StringPtr("Microsoft.Compute/virtualMachines/extensions"),
+		Name:     to.StringPtr("winrm"),
```

Comment:
actually you can probably drop this Name field here. You're already specifying it all the way at the end of this method win `CreateOrUpdate`, API will just use that name.